datumaro/plugins/image_zip_format.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import logging as log
+import os
+import os.path as osp
+from zipfile import ZIP_BZIP2, ZIP_DEFLATED, ZIP_LZMA, ZIP_STORED, ZipFile
+
+from datumaro.components.converter import Converter
+from datumaro.components.extractor import (DatasetItem, Importer,
+                                           SourceExtractor)
+from datumaro.util.image import (IMAGE_EXTENSIONS, ByteImage, decode_image,
+                                 encode_image)
+
+class ImageZipPath:
+    DEFAULT_ARCHIVE_NAME = 'default.zip'
+    DEFAULT_COMPRESSION = ZIP_STORED
+
+    COMPRESSION = {
+       'ZIP_STORED': ZIP_STORED,
+       'ZIP_DEFLATED': ZIP_DEFLATED,
+       'ZIP_BZIP2': ZIP_BZIP2,
+       'ZIP_LZMA': ZIP_LZMA
+    }
+
+class ImageZipExtractor(SourceExtractor):
+    def __init__(self, url, subset=None):
+        super().__init__(subset=subset)
+
+        assert url.endswith('.zip'), url
+
+        with ZipFile(url, 'r') as zf:
+            for path in zf.filelist:
+                item_id, extension = osp.splitext(path.filename)
+                if extension.lower() not in IMAGE_EXTENSIONS:
+                    continue
+                print(item_id)
+                image = decode_image(zf.read(path.filename))
```

Comment:
I think, having images decoded will quickly lead to memory overflow in case of a large dataset. Also, this won't allow to save images unchanged in the case of jpeg. Use `ByteImage`.